bluesky_kafka/__init__.py

```diff
@@ -486,21 +546,36 @@ class SerializerFactory(dict):
         Like a defaultdict, but it makes a Serializer based on the
         key, which in this case is the topic name.
         """
+
         def __init__(self, mongo_uri, auth_source):
             self._mongo_uri = mongo_uri
             self._auth_source = auth_source
 
         def get_database(self, topic):
-            return topic.replace('.', '-')
+            return topic.replace(".", "-")
 
         def __missing__(self, topic):
-            result = self[topic] = Serializer(self._mongo_uri + '/' + self.get_database(topic) + '?authSource=' + self._auth_source,
-                                              self._mongo_uri + '/' + self.get_database(topic) + '?authSource=' + self._auth_source)
+            result = self[topic] = mongo_normalized.Serializer(
+                self._mongo_uri
+                + "/"
+                + self.get_database(topic)
+                + "?authSource="
+                + self._auth_source,
+                self._mongo_uri
+                + "/"
+                + self.get_database(topic)
+                + "?authSource="
+                + self._auth_source,
+            )
             return result
 
-    def __init__(self, mongo_uri, auth_source='admin', *args, **kwargs):
+    def __init__(self, mongo_uri, auth_source="admin", *args, **kwargs):
         self._serializers = self.SerializerFactory(mongo_uri, auth_source)
-        return super().__init__(*args, **kwargs)
+        super().__init__(*args, **kwargs)
 
-    def process_document(self, topic, name, doc):
+    def process_document(self, consumer, topic, name, doc):
```

Comment:
Do we need to pass consumer here? I think that it is not needed if we inherit from BlueskyConsumer. Because the commit for the stop doc would happen in the base class.

Reply:
If BlueskyConsumer is constructed with a `process_document` function then the only way to send the underlying consumer to that `process_document` is as a function parameter.

Reply:
An unrelated issue is that I don't think committing just the stop document message is as helpful as committing each message as it gets processed, which I would rather leave up to client code.

Reply:
But if RemoteDispatcher inherits from BlueskyConsumer, then RemoteDispatcher would have self._consumer, and it wouldn't have to be passed.

Reply:
I should have done it a little differently, maybe like this:

```python
class BlueskyConsumer:
...
    # BlueskyConsumer.process_document() does not take a consumer argument
    def process_document(self, topic, name, doc):
        if self._process_document is None:
            ...
        else:
            # self._process_document() does take a consumer argument
            self._process_document(self._consumer, topic, name, doc)
```

Reply:
Maybe if someone wants to change how BlueskyKafka does its commits they should just make a sub class? Then we wouldn't have to pass self._consumer.

Reply:
I want `BlueskyConsumer._process_document` to be an alternative to subclassing.

Reply:
I think inheriting from Dispatcher would work just fine.  Inheriting from BlueskyConsumer just feels more organized to me, because when I think of RemoteDispatcher I think of it as a kafka Consumer that consumes Bluesky documents has callback registry. Maybe it could be named something like CallbackConsumer.  It's mainly just a personal preference about aesthetics, so I'm good with whatever you want to do.  I think aesthetics can be somewhat important, like picking good variable names.  Why do you prefer inheriting from Dispatcher instead of BlueskyConsumer?

Reply:
The problem with aesthetics is that reasonable people can disagree about them. I think of myself as being very concerned with code aesthetics, but if our principles are different then our aesthetics are as well.

The intention of bluesky_kafka.RemoteDispatcher is that it works anywhere a Dispatcher works, not anywhere a BlueskyConsumer works. The fact that it gets documents from Kafka is not part of its interface just part of its behavior.

The name "RemoteDispatcher" is not as informative as something like KafkaDispatcher. I think the intention for the original RemoteDispatcher was that the package name identified the kind of RemoteDispatcher, which was bluesky.callbacks.zmq.RemoteDispatcher. I'm not entirely satisfied with that scheme but I went along with it for the bluesky_kafka.RemoteDispatcher. The only reason for it I can think of is that changing `import RemoteDispatcher from ...` could change the RemoteDispatcher being used, but that does not seem useful. Maybe it would be better to use KafkaDispatcher. That name is very informative.